pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go

```diff
@@ -204,3 +263,28 @@ func getPreScoreState(cycleState *framework.CycleState) (*preScoreState, error)
 	}
 	return s, nil
 }
+
+func getPodRequiredNodeSelectorAndAffinity(pod *v1.Pod) (labels.Selector, *v1.NodeAffinity, error) {
+	var selector labels.Selector
+	if len(pod.Spec.NodeSelector) > 0 {
+  	selector = labels.SelectorFromSet(pod.Spec.NodeSelector)
+	}
+	var affinity *v1.NodeAffinity
+	if pod.Spec.Affinity != nil {
+		affinity = pod.Spec.Affinity.NodeAffinity
```

Comment:
use `nodeaffinity.NewLazyErrorNodeSelector` to store the parsed affinity.

and perhaps return a `preScoreState` object directly